examples/flight_replay/flight_replay.py

```diff
@@ -7,122 +7,178 @@
+
+#Start SITL if no connection string specified
+if args.connect:
+    connection_string = args.connect
+    sitl = None
+else:
+    start_lat = messages[0].lat/1.0e7
+    start_lon = messages[0].lon/1.0e7
+
+    (sitl, connection_string) = start_default_sitl()
 
-    print "Generating %s waypoints from replay..." % len(messages)
-    cmds = vehicle.commands
-    cmds.clear()
-    for i in xrange(0, len(messages)):
-        pt = messages[i]
-        lat = pt['lat']
-        lon = pt['lon']
-        # To prevent accidents we don't trust the altitude in the original flight, instead
-        # we just put in a conservative cruising altitude.
-        altitude = 30.0 # pt['alt']
-        cmd = Command( 0,
-                       0,
-                       0,
-                       mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT,
-                       mavutil.mavlink.MAV_CMD_NAV_WAYPOINT,
-                       0, 0, 0, 0, 0, 0,
-                       lat, lon, altitude)
-        cmds.add(cmd)
-    #Upload clear message and command messages to vehicle.
-    cmds.upload()
+# Connect to the Vehicle
+print 'Connecting to vehicle on: %s' % connection_string
+vehicle = connect(connection_string, wait_ready=True)
 
 
 # Now download the vehicle waypoints
 cmds = vehicle.commands
 cmds.wait_ready()
-mission_id = int(args.mission_id)
-max_freq = 0.1
-json = download_messages(mission_id, max_freq)
-print "JSON downloaded..."
-replay_mission(json)
 
 
-#Here you could actually replay the mission by
-#taking off and changing to AUTO mode.
+cmds = vehicle.commands
+cmds.clear()
+for i in xrange(0, len(messages)):
+    pt = messages[i]
+    print "Point: %d %d\n" % (pt.lat, pt.lon,)
+    lat = pt.lat
+    lon = pt.lon
+    # To prevent accidents we don't trust the altitude in the original flight, instead
+    # we just put in a conservative cruising altitude.
+    altitude = 30.0/100.0
+    cmd = Command( 0,
+                   0,
+                   0,
+                   mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT,
+                   mavutil.mavlink.MAV_CMD_NAV_WAYPOINT,
+                   0, 0, 0, 0, 0, 0,
+                   lat/1.0e7, lon/1.0e7, altitude)
+    cmds.add(cmd)
+
+#Upload clear message and command messages to vehicle.
+print("Uploading %d waypoints to vehicle..." % len(messages))
+cmds.upload()
+
+# Set mode to STABILISE for arming:
+vehicle.mode = VehicleMode("STABILIZE")
+
+while (vehicle.mode.name != "STABILIZE"):
+    time.sleep(0.1)
+
+print("Arming vehicle")
+vehicle.armed = True
+
+while not vehicle.armed:      
+    print "Waiting for arming..."
+    time.sleep(1)
+
+print "Starting mission"
+
+# Reset mission set to first (0) waypoint
+vehicle.commands.next=0
+
+# Set mode to AUTO to start mission:
+vehicle.mode = VehicleMode("AUTO")
+
+while (vehicle.mode.name != "AUTO"):
+    time.sleep(0.1)
+
+print("Raising throttle to trigger mission start")
+vehicle.channels.overrides = {'3':1500}
```

Comment:
That's actually the way you do it on a real copter!

I thought it would be good to show some variety as to how you can do things.

You disagree?